meson.build

```diff
@@ -129,10 +129,20 @@ conf.set('HAVE_STDBOOL_H', cc.has_header('stdbool.h'))
 conf.set('HAVE_MEMORY_H', cc.has_header('memory.h'))
 
 # Functions
-conf.set('HAVE_ALIGNED_ALLOC', cc.has_function('aligned_alloc', prefix: '#include <stdlib.h>'))
-conf.set('HAVE__ALIGNED_MALLOC', cc.has_function('_aligned_malloc', prefix: '#include <malloc.h>'))
-conf.set('HAVE_MEMALIGN', cc.has_function('memalign', prefix: '#include <stdlib.h>\n#include <malloc.h>'))
-conf.set('HAVE_POSIX_MEMALIGN', cc.has_function('posix_memalign', prefix: '#include <stdlib.h>'))
+if cc.has_function('memalign', prefix: '#include <stdlib.h>\n#include <malloc.h>')
+  conf.set('HAVE_MEMALIGN', 1)
+elif cc.has_function('_aligned_malloc', prefix: '#include <malloc.h>')
+  conf.set('HAVE__ALIGNED_MALLOC', 1)
+elif not (host_system == 'windows')
+  # These tests have proven unreliable on MSYS2 because meson falls
+  # back to testing compiler builtins instead of what we asked for.
```

Comment:
Also, this will be fixed in the next release with https://github.com/mesonbuild/meson/pull/1150.